src/remote.c

```diff
@@ -108,6 +109,15 @@ static int write_add_refspec(git_repository *repo, const char *name, const char
 	if ((error = ensure_remote_name_is_valid(name)) < 0)
 		return error;
 
+	if ((error = git_refspec__parse(&spec, refspec, fetch)) < 0) {
+		if (giterr_last()->klass != GITERR_NOMEMORY)
```

Comment:
I sort of wonder if we shouldn't have a function to deal with this for us...  That's more just thinking out loud than an explicit complaint.